src/main/java/ch/njol/skript/expressions/ExprLastDamage.java

```diff
@@ -19,47 +19,34 @@
  */
 package ch.njol.skript.expressions;
 
-import org.bukkit.entity.LivingEntity;
-import org.bukkit.event.Event;
-import org.eclipse.jdt.annotation.Nullable;
-
 import ch.njol.skript.classes.Changer.ChangeMode;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.expressions.base.SimplePropertyExpression;
-import ch.njol.skript.lang.Expression;
-import ch.njol.skript.lang.SkriptParser;
-import ch.njol.util.Kleenean;
 import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
 
 @Name("Last Damage")
 @Description("The last damage that was done to an entity. Note that changing it doesn't deal more/less damage.")
 @Examples({"set last damage of event-entity to 2"})
 @Since("2.5.1")
 public class ExprLastDamage extends SimplePropertyExpression<LivingEntity, Number> {
-	
+
 	static {
 		register(ExprLastDamage.class, Number.class, "last damage", "livingentities");
 	}
-	
-	@Nullable
-	private ExprDamage damageExpr;
-	
-	@Override
-	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
-		damageExpr = new ExprDamage();
-		return true;
-	}
-	
+
 	@Nullable
 	@Override
 	@SuppressWarnings("null")
 	public Number convert(LivingEntity livingEntity) {
-		return damageExpr.get(livingEntity.getLastDamageCause())[0];
```

Comment:
what kind of error was occurring here? it looks like it just wants to mimic the behavior of how `the damage` returns values, which while strange doesn't seem entirely incorrect?

Reply:
the error was because getExpr() was never set in init